frontend/src/metabase/query_builder/components/view/QuestionDataSource.unit.spec.js

```diff
@@ -282,7 +230,16 @@ class ErrorBoundary extends Component {
 }
 const SOURCE_CARD = createMockCard({ id: SOURCE_QUESTION_ID });
 
-function setup({ question, subHead = false, isObjectDetail = false } = {}) {
+function setup({
+  card,
+  subHead = false,
+  isObjectDetail = false,
+  hasPermissions = true,
+  isMultiSchemaDB = false,
+} = {}) {
+  const metadata = hasPermissions ? getMetadata() : {};
+  const question = card && new Question(card, metadata);
```

Comment:
This is the main change.
We're constructing both the `metadata` and the `question` inside of the setup. This gives us greater control.